tests/rptest/tests/random_node_operations_test.py

```diff
@@ -276,6 +276,16 @@ def test_node_operations(self, enable_failures, num_to_upgrade,
         # tp-workload-compaction - topic with compaction
         # tp-workload-fast       - topic with fast partition movements enabled
 
+        def enable_fast_partition_movement():
```

Comment:
nit: `highest_from_prior_feature_version` does some network calls when running tests locally. If this takes a bit of time it might be worth rewriting as

```py
if not with_tiered_storage:
  return False

if num_to_upgrade == 0:
  return True

initial_version = ...
supported_by_prev = ...

return supported_by_prev
```